web-app/django/VIM/apps/instruments/views/instrument_list.py

```diff
@@ -114,79 +125,138 @@ def get_context_data(self, **kwargs):
         except Exception as e:
             logger.error(f"Failed to connect to Solr: {e}")
             raise
 
-    def _get_solr_search_params(self, search_query: str, language: Language):
-        """Get common Solr search parameters."""
+    def _build_solr_query(self, language: Language):
+        """Build Solr query parameters supporting combined search + HBS filtering."""
         lang_code = language.wikidata_code
         name_field = f"instrument_name_{lang_code}_ss"
-        return {
-            "q": search_query,
+
+        # Get request parameters
+        search_query = self.request.GET.get("query", "").strip()
+        hbs_facet = self.request.GET.get("hbs_facet", None)
+
+        # Build main query (q parameter)
+        if search_query:
+            # Text search query
+            main_query = search_query
+        else:
+            # Show all instruments when no text search
+            main_query = "*:*"
+
+        # Build filter queries (fq parameter) for HBS classification
+        filter_queries = []
+        if (
+            hbs_facet and hbs_facet != EMPTY_HBS_CATEGORY
+        ):  # Used for empty/unknown categories
+            filter_queries.append(f"hbs_prim_cat_s:{hbs_facet}")
+
+        params = {
+            "q": main_query,
             "wt": "json",
             "facet": "false",
             "fl": f"sid, {name_field}, hornbostel_sachs_class_s, mimo_class_s, thumbnail_url",
             "lang_code": lang_code,
         }
 
-    def _get_solr_total_count(self, solr, search_query: str):
-        """Get total count of Solr search results."""
+        # Add filter queries if any
+        if filter_queries:
+            params["fq"] = filter_queries
+
+        return params
+
+    def _get_solr_total_count(self, solr, query_params: dict):
+        """Get total count of Solr search results with filter queries."""
         try:
             count_params = {
-                "q": search_query,
+                "q": query_params["q"],
                 "wt": "json",
                 "rows": 0,  # We only want the count
                 "facet": "false",
             }
+            # Include filter queries if present
+            if "fq" in query_params:
+                count_params["fq"] = query_params["fq"]
+
             count_response = solr.search(**count_params)
             return count_response.hits
         except Exception as e:
-            logger.error(f"Failed to get Solr count for query '{search_query}': {e}")
+            logger.error(
+                f"Failed to get Solr count for query '{query_params['q']}': {e}"
+            )
             return 0
 
     def _get_solr_page_results(
-        self, solr, search_params: dict, page_size: int, start: int
+        self, solr, query_params: dict, page_size: int, start: int
     ):
-        """Get a specific page of Solr search results."""
+        """Get a specific page of Solr search results with filter queries."""
         try:
```

Comment:
Only put the stuff in the try / except that will actually raise an exception, and only catch the exceptions that you think will be raised. Otherwise you're hiding errors that nobody will see, they'll just see a broken website.